File: Backend/app/models/canteen_model.py
from datetime import datetime
from pydantic import BaseModel
from typing import List, Optional
from enum import Enum

class CanteenStatusEnum(str, Enum):
    requested = 'requested'
    approved = 'approved'
    rejected = 'rejected'

class Canteen(BaseModel):
    Name : str
    Id : int
    OwnerId : int
    Address : str
    Latitude : Optional[float] = None
    Longitude : Optional[float] = None
    Description : Optional[str] = None
    WorkingHours : str
    AverageRating: Optional[float] = None
    CreatedTime: Optional[datetime] = None
    LastUpdated: Optional[datetime] = None
    Status: Optional[CanteenStatusEnum] = None
    Deleted: Optional[bool] = False
    

class CanteenRegistration(BaseModel):
    Id: int
    Created: datetime
    OwnerId: int
    CanteenId: int
    status: CanteenStatusEnum
    Comments: Optional[str] = None
    

File: Frontend/UNCGEats/src/app/models/canteen.model.ts
import { User } from "./user.model"

export interface Canteen{
    Name : string
    Id?: number
    OwnerId: number
    Address : string
    Latitude? : number
    Longitude? : number
    Description? : string
    WorkingHours : string
    AverageRating: number
    CreatedTime?: Date
    LastUpdated?: Date
    Image?: string
    _id?: string
    Deleted?: boolean
    Status? : CanteenStatus
    Comments? : string
}


export enum CanteenStatus {
    requested = 'Requested',
    approved = 'Approved',
    rejected = 'Rejected'
}

export interface CanteenRegistration {
  Id?: number;
  Created?: Date;
  OwnerId: number;
  CanteenId: number;
  Status: CanteenStatus;
  Comments?: string;
  LastUpdated?: Date;
  Canteen?: Canteen;
  Owner?: User;
}

